tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp

```diff
@@ -250,6 +253,369 @@ TEST (nnstreamer_nnfw_runtime_raw_functions, invoke)
+
+  /* Open model with proper dimension */
+  status = ml_single_open (&single, test_model, in_info, out_info,
+      ML_NNFW_TYPE_NNFW, ML_NNFW_HW_ANY);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  /* input tensor in filter */
+  status = ml_single_get_input_info (single, &in_res);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  status = ml_tensors_info_get_count (in_res, &count);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_EQ (count, 1U);
+
+  status = ml_tensors_info_get_tensor_type (in_res, 0, &type);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_EQ (type, ML_TENSOR_TYPE_FLOAT32);
+
+  ml_tensors_info_get_tensor_dimension (in_res, 0, res_dim);
+  EXPECT_TRUE (in_dim[0] == res_dim[0]);
+  EXPECT_TRUE (in_dim[1] == res_dim[1]);
+  EXPECT_TRUE (in_dim[2] == res_dim[2]);
+  EXPECT_TRUE (in_dim[3] == res_dim[3]);
+
+  input = output = NULL;
+
+  /* Change and update dimension for mismatch */
+  in_dim[0] = in_dim[1] = in_dim[2] = in_dim[3] = 2;
+  ml_tensors_info_set_tensor_dimension (in_info, 0, in_dim);
+
+  status = ml_tensors_data_create (in_info, &input);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_TRUE (input != NULL);
+
+  status = ml_tensors_data_get_tensor_data (input, 0, (void **) &data, &data_size);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_EQ (data_size, sizeof (float)*16);
+  data[0] = 10.0;
+
+  status = ml_single_invoke (single, input, &output);
+  EXPECT_EQ (status, ML_ERROR_INVALID_PARAMETER);
+
+  ml_tensors_data_destroy (output);
+  ml_tensors_data_destroy (input);
+
+  status = ml_single_close (single);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+
+  g_free (test_model);
+  ml_tensors_info_destroy (in_info);
+  ml_tensors_info_destroy (out_info);
+  ml_tensors_info_destroy (in_res);
+}
+
+/**
+ * @brief Callback for tensor sink signal.
+ */
+static void
+new_data_cb (const ml_tensors_data_h data, const ml_tensors_info_h info, void *user_data)
+{
+  int status;
+  float *data_ptr;
+  size_t data_size;
+
+  status = ml_tensors_data_get_tensor_data (data, 0, (void **) &data_ptr, &data_size);
+  EXPECT_EQ (status, ML_ERROR_NONE);
+  EXPECT_EQ (*data_ptr, 12.0);  
+}
+
+/**
+ * @brief Test nnfw subplugin with successful invoke (pipeline, ML-API)
+ */
+TEST (nnstreamer_nnfw_mlapi, invoke_pipeline_00)
+{
+  gchar *pipeline;
+  ml_pipeline_h handle;
+  ml_pipeline_src_h src_handle;
+  ml_pipeline_sink_h sink_handle;
+  ml_tensor_dimension in_dim;
+  ml_tensors_info_h info;
+  ml_pipeline_state_e state;
+  ml_tensors_data_h input;
+  float *data;
+  size_t data_size;
+  
+  gchar *test_model;
+
+  test_model = get_model_file ();
+  ASSERT_TRUE (test_model != nullptr);
+
+  pipeline =
+      g_strdup_printf ("appsrc name=appsrc ! "
+      "other/tensor,dimension=(string)1:1:1:1,type=(string)float32,framerate=(fraction)0/1 ! "
+      "tensor_filter framework=nnfw model=%s ! "
+      "tensor_sink name=tensor_sink", test_model);
+
+  int status = ml_pipeline_construct_internal (pipeline, NULL, NULL, &handle);
```

Comment:
plz use ml_pipeline_construct(), not _internal()

Reply:
I updated what you mentioned.